app/api/cron/dune/program-calls/route.ts

```diff
@@ -0,0 +1,67 @@
+import { headers } from 'next/headers';
+import { db } from '@/src/db/drizzle';
+import { program_call_stats } from '@/src/db/schema';
+import { NextResponse } from 'next/server';
+import { DuneClient, RunQueryArgs } from '@duneanalytics/client-sdk';
+import { Cluster } from '@utils/cluster';
+import { fetchProgramMetadataIdl, programNameFromIdl } from '@/app/components/instruction/codama/getProgramMetadataIdl';
+import { PROGRAM_INFO_BY_ID } from '@/app/utils/programs';
+
+const { DUNE_API_KEY, DUNE_PROGRAM_CALLS_MV_ID } = process.env;
+
+export async function GET() {
+    const headersList = await headers();
+    if (headersList.get('Authorization') !== `Bearer ${process.env.CRON_SECRET}`) {
+        return NextResponse.json({ error: 'Unauthorized' }, { status: 401 });
+    }
+
+    const client = new DuneClient(DUNE_API_KEY ?? '');
+    const opts: RunQueryArgs = { queryId: Number(DUNE_PROGRAM_CALLS_MV_ID) };
+    const executionResult = await client.getLatestResult(opts);
+
+    await db.transaction(async tx => {
+        await tx.delete(program_call_stats).execute();
+
+        const values = await Promise.all(
+            (executionResult.result?.rows ?? []).map(async row => ({
+                program_address: String(row.program_address),
+                name: await buildProgramName(row),
+                description: String(row.program_description),
+                address: String(row.address),
+                calls_number: Number(row.calls_number),
+            }))
+        );
+
+        await tx.insert(program_call_stats).values(values).execute();
+    });
+
+    return NextResponse.json({ ok: true });
+}
+
+async function buildProgramName(row: Record<string, any>): Promise<string> {
+    if (PROGRAM_INFO_BY_ID[row.address]) {
+        return String(PROGRAM_INFO_BY_ID[row.address].name);
+    }
+    const pmName = await getPmName(String(row.address));
+    if (pmName !== null && pmName !== undefined && pmName !== '') {
+        return String(pmName);
+    }
+    return String(row.program_name);
+}
+
+async function getPmName(address: string): Promise<string> {
+    const idl = await fetchProgramMetadataIdl(address, 'https://api.mainnet-beta.solana.com', Cluster.MainnetBeta);
```

Comment:
I see a couple of issues here:
1. We do not use env. variable for endpoint
On production, this endpoint will likely rate-limit the requests. Explorer uses a custom endpoint, and we need to reuse it
2. In theory, `fetchProgramMetadataIdl` might throw an error, and that leads to idl being null. But it also returns null if PMP_IDL_ENABLED is disabled. That means we can not differentiate actual error from the case where the endpoint is turned off.

Reply:
There is a `serverClusterUrl` helper that handles url for the cluster properly